File: day04/src/main/scala/de/havox_design/aoc2017/day04/Day04.scala
package de.havox_design.aoc2017.day04

object Day04 {
  def solvePart1(filename: String): Int =
    readData(filename)
      .count{ row =>
        val rowElements = row.split(" ")
        rowElements.distinct.length == rowElements.length
      }

  def solvePart2(filename: String): Int =
    0

  def readData(filename: String): Iterator[String] =
    scala.io.Source.fromResource(filename).getLines()
}

File: aoc2021/day04/src/main/java/de/havox_design/aoc2016/day04/Day04.java
package de.havox_design.aoc2016.day04;

import de.havox_design.aoc.utils.java.input.DataReader;

import java.util.List;

public class Day04 {
    private final List<String> input;

    public Day04(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day04 instance = new Day04(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day04 instance = new Day04(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}
